File: src/Models/Response/SimClassResponse.cs
using System.Text.Json.Serialization;
using Models.Enums;

namespace Models.Response;

public class SimClassResponse()
{
    public Guid? Id { get; set; }
    public string? Name { get; set; }
    public Guid? IdBaseClass { get; set; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public SimModelsAccesibility State { get; set; }
    public List<MethodResponse> Methods { get; set; } = [];
    public List<AttributeResponse> Attributes { get; set; } = [];
    public List<InterfaceResponse> Implements { get; set; } = [];
    public NamespaceResponse? BaseNamespace { get; set; }
}

File: frontend/src/app/models/response/SimClassResponse.ts
import { MethodResponse } from './MethodResponse';
import { AttributeResponse } from './AttributeResponse';
import { NamespaceResponse } from './NamespaceResponse';
import { InterfaceResponse } from './InterfaceResponse.model';

export interface SimClassResponse {
    id?: string;
    name?: string;
    idBaseClass?: string;
    state: string;
    methods: MethodResponse[];
    attributes: AttributeResponse[];
    implements: InterfaceResponse[];
    namespace?: NamespaceResponse;
}